.github/workflows/lint.yml

```diff
@@ -63,11 +64,16 @@ jobs:
           tox -e ruff -- check
 
       - name: Run linting
-        if: success() || failure()
+        if: ${{ !cancelled() && (steps.deps.outcome == 'success') }}
```

Comment:
Apparently, `cancelled` is preferred.

```
Warning: Avoid using always for any task that could suffer from a critical failure, for example: getting sources, otherwise the workflow may hang until it times out. If you want to run a job or step regardless of its success or failure, use the recommended alternative: if: ${{ !cancelled() }}
```

from [here](https://docs.github.com/en/actions/learn-github-actions/expressions)